packages/waku/src/lib/builder/constants.ts

```diff
@@ -5,3 +5,4 @@ export const DIST_SERVE_JS = 'serve.js';
 export const DIST_PUBLIC = 'public';
 export const DIST_ASSETS = 'assets';
 export const DIST_SSR = 'ssr';
+export const PERIOD_ESCAPE = ':mime:';
```

Comment:
Is this a Vite thing?

Reply:
No, this is so that `extname` can be called without matching on `'.'` in the slug folder name. 

As part of the waku build process I am temporarily replacing `.` with this arbitrary constant (code change in `packages/waku/src/router/create-pages.ts`), so that we can call `extname` and do all the other normal logic, then just clean up the temporary symbol at the end (like 598 in `packages/waku/src/lib/builder/build.ts`)

Reply:
Another solution I can suggest is just disallowing `.` in slugs. This could be done with a console warning and removing the `.` altogether instead of replacing it with an escape string to add back in at the end.

Reply:
Ok, I actually think disallowing `.` in slugs and simply removing them will be the best strategy here. 

This is the build output when doing the escaping I have currently in this PR:

<img width="467" alt="image" src="https://github.com/user-attachments/assets/5d7bc831-f4fe-4fa0-9b77-89c756ce9e80">

And this is what I would expect the output to be if `.` was considered valid in a slug. However, the route `http://localhost:8080/mr. mime` results in a 404 and is unreachable. It could at that point be an issue in `http-server`, but I can't honestly think of a url path that used `.` as part of the path (it is always for the extension).

I'll add the commit to just strip out `.` from slugs, but happy to change if you'd rather another strategy.

Reply:
Along these same lines: Should `' '` be replaced with `'-'`?

Reply:
Tested with Astro (best standard I can think of for handling SSG like this)

for a file named `mr. mime.md` it gives the slug: `mr-mime`

<img width="524" alt="image" src="https://github.com/user-attachments/assets/3f011809-69de-4fe9-a6e9-36be834f6299">

I will update to do the following to match this
- remove `.` characters from slugs
- replace ` ` with `-`

Reply:
> I will update to do the following to match this
> 
> * remove `.` characters from slugs
> * replace ` ` with `-`

Sounds good. Can you also add this current spec to https://github.com/dai-shi/waku/blob/main/docs/create-pages.mdx?